File: src/main/kotlin/io/andrelucas/business/Toilet.kt
package io.andrelucas.business

import java.util.UUID

data class Toilet(val id: UUID, val name:String, val geolocation: Geolocation, val type: ToiletType){
    companion object {
        fun create(name: String, geolocation: Geolocation, type: ToiletType): Toilet {
            return Toilet(UUID.randomUUID(), name, geolocation, type)
        }
    }
}

File: src/main/scala/toilet/Toilet.scala
package io.andrelucas

import io.andrelucas.toilet.ToiletInvalidException

import java.util.UUID

case class Toilet(id: UUID,
                  description: String,
                  geolocation: Geolocation,
                  items: List[Items],
                  genre: List[Genre])

object Toilet:

  type RegisterToilet = (String, Double, Double)
  def register(registerToilet: RegisterToilet): Either[Throwable, Toilet] =
    if registerToilet._1.isEmpty then 
      Left(ToiletInvalidException("Toilet does not have a valid description")) 
    else 
      for {
        geo <- Geolocation(registerToilet._2, registerToilet._3)
      } yield Toilet(UUID.randomUUID(), registerToilet._1, geo, List.empty[Items], List.empty[Genre])



//    Geolocation(registerToilet._2, registerToilet._3)
//      .flatMap { r =>
//        Right(Toilet(UUID.randomUUID(), registerToilet._1, r, List.empty[Items], List.empty[Genre]))
//      }
